src/engine/test/source/hlp/hlp_test.cpp

```diff
@@ -241,8 +233,56 @@ TEST(hlpTests_IPaddress, IPV6_failed)
     ASSERT_TRUE(result.find("_ip") == result.end());
 }
 
-// Test: parsing json objects
-TEST(hlpTests_json, success_parsing)
+// Test: parsing json
+TEST(hlpTests_json, parameters_failure_cases)
+{
+    const char* logQl1 = "<_json/json/param1/param2>";
+    const char* logQl2 = "<_json/json/wrongType>";
+
+    const char* event = "{\"key1\":\"value1\",\"key2\":\"value2\"}";
+
+    ASSERT_THROW(getParserOp(logQl1), std::runtime_error);
+    ASSERT_THROW(getParserOp(logQl2), std::runtime_error);
+}
+
+TEST(hlpTests_json, object_success)
+{
+    const char* logQl = "<_json/json/object>";
+
+    const char* event = "{\"key1\":\"value1\",\"key2\":\"value2\"}";
+
+    auto parseOp = getParserOp(logQl);
+    ParseResult result;
+    bool ret = parseOp(event, result);
+
+    ASSERT_EQ(true, static_cast<bool>(parseOp));
+    ASSERT_EQ("{\"key1\":\"value1\",\"key2\":\"value2\"}",
+              std::any_cast<JsonString>(result["_json"]).jsonString);
```

Comment:
Why not ASSERT_SRTEQ to compare string?
A code like the following would be much easier to read and write:
```CPP
    ASSERT_TRUE(static_cast<bool>(parseOp));
    const auto expectedResult {R"({"key1":"value1","key2":"value2"})"};
    ASSERT_STREQ(expectedResult,
                 std::any_cast<JsonString>(result["_json"]).jsonString.data());
```

Reply:
got it